python/lsst/display/firefly/firefly.py

```diff
@@ -89,13 +88,20 @@ def __init__(self, display, verbose=False, host="http://localhost:8080",
         global _fireflyClient
         if not _fireflyClient:
             try:
-                _fireflyClient = firefly_client.FireflyClient(host,
-                                        channel=name, basedir=basedir, **kwargs)
+                if url is None:
+                    _fireflyClient = firefly_client.FireflyClient(channel=name, **kwargs)
+                else:
+                    _fireflyClient = firefly_client.FireflyClient(url,
+                                        channel=name, **kwargs)
             except Exception as e:
```

Comment:
Typically it's better to trap the exceptions you expect.  Trapping a bare exception means you can't tell what actually went wrong when something unexpected happens.

Reply:
Agreed. I'll trap on the two exceptions that are raised, one for a nonsense url and one for a working web server that is not a Firefly url.